### PR TITLE
ADS: Custom Builders for AlertDialog components

### PR DESCRIPTION
**Pull request by nalcalag**
Task/Issue URL: https://app.asana.com/0/0/1203198140265639/f

### Description

- Create custom builders for each dialog component: `TextAlertDialog`, `StackedAlertDialog` and `RadioListAlertDialog`
- Moved dialog background color to style from rootView

### Steps to test this PR

- Install from this branch
- Go to Settings > App Components Design Preview
- Tap on "Dialogs" tab
- [ ] Check all dialogs under DIALOGS title look fine

### CODE REVIEW

**Comment by nalcalag** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2436** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2436" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2436?utm_source=stack-comment).

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/RadioListAlertDialog.kt [hunk 1]
```diff
@@ -16,20 +16,17 @@
...
+import androidx.appcompat.app.AlertDialog
 import com.duckduckgo.mobile.android.databinding.DialogSingleChoiceAlertBinding
 import com.duckduckgo.mobile.android.ui.view.button.RadioButton
-import com.duckduckgo.mobile.android.ui.view.dialog.StackedAlertDialog.Builder
 import com.duckduckgo.mobile.android.ui.view.gone
 import com.duckduckgo.mobile.android.ui.view.toDp
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 
-class RadioListAlertDialog(val builder: Builder) : DialogFragment() {
+class RadioListAlertDialog {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/RadioListAlertDialog.kt:29]: we don't need this class anymore

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/RadioListAlertDialog.kt [hunk 2]
```diff
@@ -41,153 +38,138 @@ class RadioListAlertDialog(val builder: Builder) : DialogFragment() {
...
+            val dialogBuilder = MaterialAlertDialogBuilder(context, com.duckduckgo.mobile.android.R.style.Widget_DuckDuckGo_Dialog)
+                .setView(binding.root)
+                .apply {
+                    setCancelable(false)
+                    setOnDismissListener { listener.onDialogDismissed() }
+                }
+            val dialog = dialogBuilder.create()
+
+            setViews(binding, dialog)
+            checkRequiredFieldsSet()
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/RadioListAlertDialog.kt:115]: this should be done at the beginning of the method, so we avoid inflating layouts if the required fields are missing

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/StackedAlertDialog.kt [hunk 3]
```diff
@@ -16,20 +16,18 @@
...
 import androidx.annotation.StringRes
-import androidx.fragment.app.DialogFragment
+import androidx.appcompat.app.AlertDialog
 import com.duckduckgo.mobile.android.databinding.DialogStackedAlertBinding
 import com.duckduckgo.mobile.android.ui.view.button.ButtonGhostLarge
 import com.duckduckgo.mobile.android.ui.view.gone
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 
-class StackedAlertDialog(val builder: Builder) : DialogFragment() {
+class StackedAlertDialog {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/StackedAlertDialog.kt:30]: same as above, we don't need this class anymore

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/StackedAlertDialog.kt [hunk 4]
```diff
@@ -38,133 +36,117 @@ class StackedAlertDialog(val builder: Builder) : DialogFragment() {
...
+            val dialogBuilder = MaterialAlertDialogBuilder(context, com.duckduckgo.mobile.android.R.style.Widget_DuckDuckGo_Dialog)
+                .setView(binding.root)
+                .apply {
+                    setCancelable(false)
+                    setOnDismissListener { listener.onDialogDismissed() }
+                }
+            val dialog = dialogBuilder.create()
+
+            setViews(binding, dialog)
+            checkRequiredFieldsSet()
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/StackedAlertDialog.kt:100]: same, we can move `checkRequiredFieldsSet()` to the top

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/TextAlertDialog.kt [hunk 5]
```diff
@@ -16,19 +16,16 @@
...
 import androidx.annotation.StringRes
-import androidx.fragment.app.DialogFragment
+import androidx.appcompat.app.AlertDialog
 import com.duckduckgo.mobile.android.databinding.DialogTextAlertBinding
-import com.duckduckgo.mobile.android.ui.view.dialog.StackedAlertDialog.Builder
 import com.duckduckgo.mobile.android.ui.view.gone
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 
-class TextAlertDialog(val builder: Builder) : DialogFragment() {
+class TextAlertDialog {
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/TextAlertDialog.kt:28]: ditto

File: common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/TextAlertDialog.kt [hunk 6]
```diff
@@ -39,137 +36,119 @@ class TextAlertDialog(val builder: Builder) : DialogFragment() {
...
+            val dialogBuilder = MaterialAlertDialogBuilder(context, com.duckduckgo.mobile.android.R.style.Widget_DuckDuckGo_Dialog)
+                .setView(binding.root)
+                .apply {
+                    setCancelable(false)
+                    setOnDismissListener { listener.onDialogDismissed() }
+                }
+            val dialog = dialogBuilder.create()
+
+            setViews(binding, dialog)
+            checkRequiredFieldsSet()
```
Comment (malmstein) [inline common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/TextAlertDialog.kt:105]: ditto

**Review by malmstein — APPROVED** [review]
Looks good, couple comments but great job!

**Pull request merged**